hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/handlers/FSOBucketHandler.java

```diff
@@ -146,49 +145,116 @@ public long calculateDUUnderObject(long parentId)
           break;
         }
         OmKeyInfo keyInfo = kv.getValue();
+        if (recursive) {
+          if (stats.getLimit() == -1) {
+            populateDiskUsage(keyInfo, diskUsageList);
+          } else {
+            if (stats.getCurrentCount() < stats.getLimit()) {
+              populateDiskUsage(keyInfo, diskUsageList);
+              stats.setCurrentCount(stats.getCurrentCount() + 1);
+              stats.setLastKey(kv.getKey());
```

Comment:
for list file, LastKey can be one of file, but seems current logic do not support iteration continue from that point, eg:
- lets list file for a volume /a
- limit reached for one of file /a/b/c/d/1.txt and this is last key
- How this will continue listing further keys and other bucket files if given as input ?

Shall we use other listing operation as we use from rocks db iterator ? which is lexicographic ordering ?

Reply:
@sumitagrawl , thanks for reviewing the patch. Current listKeys API has this limit parameter which will limit in order. If other buckets also present in volume and if limit reached, i will not list. This is similar to what we have in CLI behavior.